semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbJavacOptions.java

```diff
@@ -73,7 +78,11 @@ public static SemanticdbJavacOptions parse(String[] args, Context ctx) {
     if (result.targetroot == null && !useJavacClassesDir) {
       result.errors.add(missingRequiredDirectoryOption("targetroot"));
     }
-    if (result.sourceroot == null) {
+    if (result.uriScheme != UriScheme.BAZEL && result.sourceroot == null) {
```

Comment:
Would it make sense to create a small function/static method for this check:

```
result.uriScheme != UriScheme.BAZEL && result.sourceroot == null
```

It looks like the negation of this is (correctly) tested in the `inferBazelSourceroot` method, having a common name would make the connection immediately clear.

Reply:
Good point. I refactored this condition into a separate method and added comments to clarify the behavior
```java
  private static boolean isSourcerootDefined(SemanticdbJavacOptions options) {
    if (options.uriScheme == UriScheme.BAZEL) {
      return true; // The sourceroot is automatically inferred for Bazel.
    }
    return options.sourceroot != null;
  }
```